src/toplevel.c

```diff
@@ -295,43 +296,73 @@ JL_DLLEXPORT jl_module_t *jl_base_relative_to(jl_module_t *m)
     return jl_top_module;
 }
 
-int jl_has_intrinsics(jl_value_t *v)
+static void expr_attributes(jl_value_t *v, int *has_intrinsics, int *has_defs)
 {
     if (!jl_is_expr(v))
-        return 0;
+        return;
     jl_expr_t *e = (jl_expr_t*)v;
-    if (e->head == toplevel_sym || e->head == copyast_sym)
-        return 0;
-    if (e->head == foreigncall_sym)
-        return 1;
+    jl_sym_t *head = e->head;
+    if (head == toplevel_sym || head == thunk_sym) {
+        return;
+    }
+    else if (head == global_sym || head == const_sym || head == copyast_sym) {
```

Comment:
`global_sym` isn't a definition anymore, it's just a scope annotation

copyast also doesn't seem like it defines anything, it's just a local allocation (albeit one that commonly appears in contexts related to defining code)

Reply:
I don't think that's true; `global` as a scope annotation gets removed by the front end, but when it serves to create an (initially undefined) global variable the `global` expression remains.

> one that commonly appears in contexts related to defining code

Yes, that's what I was thinking.